logging/annotator.go

```diff
@@ -0,0 +1,34 @@
+package logging
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"google.golang.org/grpc/metadata"
+)
+
+// HTTP Headers
+const logLevelHeaderKey = "log-level"
+const logFlagHeaderKey = "log-trace-key"
+
+// Metadata keys used to pass from gateway to service
+const logLevelMetaKey = "log-level"
+const logFlagMetaKey = "log-trace-key"
+
+// Name of field to be logged if included
+const logFlagFieldName = "log-trace-key"
```

Comment:
This seems similar to having a custom "Request-Id" (e.g. if we can set Request-Id from outside). It is possible to specify request-id from the outside if you use request-id interceptor. Are there any other use cases for this?

Reply:
Request-ID should always be unique, this isn't necessarily unique.